### PR TITLE
feat(lnd): update BTCD and LND to latest versions

### PR DESCRIPTION
**Pull request by mrfelton**
In an ideal world, lnd would make nightly snapshots available that we could download. However currently they don't do this so I have created our own solution that we can use to compile and publish lnd binaries.

I have created a fork of the lnd repo at https://github.com/LN-Zap/lnd on which I have added support for [goreleaser](https://goreleaser.com/).

You can see the changes that enable this here:
https://github.com/lightningnetwork/lnd/compare/master...LN-Zap:zap/canary-releases?expand=1

With this, we can easily publish a new release with a simple call to `./canary.sh`. This will build lnd for all of the platforms and targets that we support and then create a github release and publish the binaries to it.

For example, here is the release that I have just created from the current master HEAD commit.
https://github.com/LN-Zap/lnd/releases

Then we can use these binaries in conjunction with lnd-binary which is what this PR does.

- [x] Publish updated lnd pre-compiled binary
- [x] Update btcd backend
- [x] Update sync process
- [x] Update grpc proto and api usage
- [ ] Set up DNS name for btcd cluster

**NOTE: This PR should only be merged as part of an update that adds support for the latest lnd grpc interface to zap (update proto + usage)**

**NOTE 2: This upgrade will require users to delete their local lnd data and do a full re-sync**

### CODE REVIEW

File: app/lib/lnd/neutrino.js [hunk 1]
```diff
@@ -55,6 +80,17 @@ class Neutrino extends EventEmitter {
       `${this.lndConfig.alias ? `--alias=${this.lndConfig.alias}` : ''}`
     ]
 
+    switch (this.lndConfig.network) {
```
Comment (JimmyMow) [inline app/lib/lnd/neutrino.js]: Why did you opt for a `switch` statement with only 2 options? Are we expecting this conditional to grow? There is also no default clause here

**Comment by JimmyMow** [comment]
@mrfelton 1 small comment and also don't we wanna change `app/components/Activity/InvoiceModal.js` to display `invoice.value` if the invoice has been settled?

File: app/lib/lnd/neutrino.js (hunk 1 above)
Comment (mrfelton) [reply app/lib/lnd/neutrino.js]: Good question. Actually at the time I had in my mind about supporting multiple different currencies but even then each one would probably only have testnet and mannet support. So I'll update this to use if/else.

**Comment by mrfelton** [comment]
@JimmyMow I figured it might be better to leave the change to the invoices for a separate PR as that change will actually break the invoice display for people connecting to lnd 0.4.2.

**Pull request merged**